src/flint/test/test.py

```diff
@@ -1584,6 +1584,52 @@ def test_pickling():
         obj2 = pickle.loads(s)
         assert obj == obj2
 
+def test_fmpz_mod():
+    from flint import fmpz_mod_ctx
+
+    # Context tests
+    F163 = fmpz_mod_ctx(163)
+    assert F163.modulus() == 163
+    big = 2**1024 - 1
+    Fbig = fmpz_mod_ctx(2**1024 - 1)
+    assert Fbig.modulus() == big
+
+    # Rich comparisons 
+    assert raises(lambda: F163(123) > 0, TypeError)
+    assert raises(lambda: F163(123) >= 0, TypeError)
+    assert raises(lambda: F163(123) < 0, TypeError)
+    assert raises(lambda: F163(123) <= 0, TypeError)
+
+    assert F163(123) == 123
+    assert F163(123) == 123 + 163
+    assert F163(123) != 1
+    assert F163(123) == F163(123)
```

Comment:
I would generally write these tests like:
```python
assert (F163(123) == 123) is True
```
The reason is that most objects in Python are "truthy" so in this case
```python
assert F163(0) == 0
```
the `F163(0).__eq__` method could return almost anything and the test would still pass.

This consideration only applies to testing the behaviour of `__eq__` and related methods though. It is important to be very careful and explicit in the `__eq__` tests because if `__eq__` doesn't work then we can't trust any of the other tests!